src/argparse.lua

```diff
@@ -1069,6 +1109,76 @@ function Parser:get_help()
    return table.concat(blocks, "\n\n")
 end
 
+local function shell_escape(string)
+   return string:gsub("[\\']", "\\%0")
+end
+
+function Parser:get_bash_complete() print "not yet implemented" end
+
+function Parser:get_zsh_complete() print "not yet implemented" end
+
+function Parser:_fish_complete_help(lines, prefix)
+   for _, command in ipairs(self._commands) do
+      for _, alias in ipairs(command._aliases) do
+         local line = ("%s -n '__fish_use_subcommand' -xa '%s\\t\"%s\"'")
+            :format(prefix, alias, shell_escape(command._description))
+         table.insert(lines, line)
+      end
+   end
+
+   for _, option in ipairs(self._options) do
+      local parts = {prefix}
+
+      if self._parent then
+         local aliases = table.concat(self._aliases, " ")
+         table.insert(parts, ("-n '__fish_seen_subcommand_from %s'"):format(aliases))
```

Comment:
I'm always scared of function calls inside of `table.insert` due to `table.insert`'s weird optional middle argument behaviour.
Could you change this to be just `parts[2] = `? or if `prefix` can be `nil`, `parts[#parts+1] = `